elfloader-tool/src/arch-arm/smp_boot.c

```diff
@@ -34,7 +34,11 @@ void non_boot_main(void)
 #endif
     /* Spin until the first CPU has finished initialisation. */
     while (!non_boot_lock) {
-#ifndef CONFIG_ARCH_AARCH64
+#ifdef CONFIG_ARCH_AARCH64
+        /* The compiler may optimize this loop away, add a dsb()
+         * to force a reload. */
+        dsb();
```

Comment:
This also applies to non-aarch64, so this doesn't look like the right solution. Using standard atomics may be better and more portable here.